core/fileproc.py

```diff
@@ -25,17 +25,36 @@ def check_file_type(file_path, file_types):
     return False
 
 
-def get_file_paths(base_path, file_types):
+def include_file(file_path, white_list_patterns):
+    """
```

Comment:
I really love this approach. The use of regex here is impressive :clap: 
I just want that you elaborate a bit in the docs and keep the style consistent. For example: 
```
def check_file_type(file_path, file_types):
    """
    check file type to assert that only file with certain predefined extensions
    are checked.
    Args:
        file_path   (str) : path to file.
        file_types (list) : list of file extensions to accept.
    Returns:
        boolean, true if file type is supported else false.
    """
    ftype = "." + file_path.split(".")[-1]
    if ftype in file_types:
        return True

    # default return
    return False
```

Reply:
yes I can definitely fix this up!